vissl/losses/barlow_twins_loss.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import pprint
+
+import torch
+from classy_vision.generic.distributed_util import get_cuda_device_index
+from classy_vision.losses import ClassyLoss, register_loss
+from torch import nn
+from vissl.utils.hydra_config import AttrDict
+
+
+@register_loss("barlow_twins_loss")
+class BarlowTwinsLoss(ClassyLoss):
+    """
+    This is the loss which was proposed in the Barlow Twins https://arxiv.org/abs/2103.03230v1
+    paper.
+    See the paper for the details on the loss.
+
+    Config params:
+        lambda_ (float): weight on the off-diagonal terms. It controls the trade-off between
+            the importance given to the invariance term versus the redundancy reduction term.
+        embedding_dim (int): dimensionality of the representation
+    """
+
+    def __init__(self, loss_config: AttrDict):
+        super(BarlowTwinsLoss, self).__init__()
+
+        self.loss_config = loss_config
+        self.bt_criterion = BarlowTwinsCriterion(
+            self.loss_config.lambda_, self.loss_config.embedding_dim
+        )
+
+    @classmethod
+    def from_config(cls, loss_config: AttrDict):
+        """
+        Instantiates BarlowTwinsLoss from configuration.
+
+        Args:
+            loss_config: configuration for the loss
+
+        Returns:
+            BarlowTwinsLoss instance.
+        """
+        return cls(loss_config)
+
+    def forward(self, output, target):
```

Comment:
can we add type hints for `output` and `target`? for example https://github.com/facebookresearch/vissl/blob/master/vissl/losses/swav_loss.py#L83